exercises/binary-search/example.cpp

```diff
@@ -0,0 +1,28 @@
+#include "binary_search.h"
+
+namespace binary_search {
+
+int find(std::vector<int> const& data, int value)
+{
+    int lpos = 0;
+    int rpos = data.size() - 1;
+    int mid = 0;
+
+    while (lpos <= rpos)
+    {
+        mid = (lpos + rpos)/2;
+        
+        if (value < data[mid]) {
+            rpos = mid - 1;
+        } else if (value > data[mid]) {
+            lpos = mid + 1;
+        }
+        
+        if (data[mid] == value) { 
+            return mid; 
+        }
+    }
+        return -1;
```

Comment:
```suggestion
        return SIZE_MAX;
```

Reply:
The updated problem spec suggests to raise a "value not in array" error in the last 5 cases. I changed my example and tests such that an exception is thrown and tested for, respectively. What are your thoughts on this?

Reply:
I think throwing is probably fine. I don't like it but it does fit the problem spec best :\